src/legacy/core_plugins/kibana/public/discover/doc_table/lib/get_sort.js

```diff
@@ -45,24 +42,33 @@ function createSortObject(sortPair, indexPattern) {
  * @returns {object} a sort object suitable for returning to elasticsearch
  */
 export function getSort(sort, indexPattern, defaultSortOrder = 'desc') {
-
   let sortObjects;
   if (Array.isArray(sort)) {
-    sortObjects = _.compact(sort.map((sortPair) => createSortObject(sortPair, indexPattern)));
+    sortObjects = _.compact(sort.map(sortPair => createSortObject(sortPair, indexPattern)));
   }
 
   if (!_.isEmpty(sortObjects)) {
     return sortObjects;
-  }
-  else if (indexPattern.timeFieldName && isSortable(indexPattern.timeFieldName, indexPattern)) {
-    return [{ [indexPattern.timeFieldName]: defaultSortOrder }];
-  }
-  else {
+  } else if (indexPattern.timeFieldName && isSortable(indexPattern.timeFieldName, indexPattern)) {
+    if (indexPattern.isTimeNanosBased()) {
+      // is necessary for sorting indices with date AND date_nanos
+      return [
+        {
+          [indexPattern.timeFieldName]: { order: defaultSortOrder, numeric_type: 'date_nanos' },
+        },
+      ];
```

Comment:
Dear valued reviewer, here's the relevant change I did, everything else was done by my pal prettier